google/cloud/opentelemetry/quickstart/README.md

```diff
@@ -0,0 +1,227 @@
+does not).
+
+## The Quickstart
+
+### What it does
+
+The quickstart creates a single trace. The root span is named "quickstart". The
+quickstart makes some routine client calls, each of which are traced. These
+calls show up as children spans of the overarching "quickstart" span. Upon exit
+of the quickstart, the collected trace is sent to Cloud Trace.
+
+### Viewing the trace in Cloud Trace
+
+To find the traces created by this quickstart, we can follow the documentation
+to [Find and view traces]. We can search using the filter `RootSpan:quickstart`
+in the Cloud Trace UI for the project ID we used to run the quickstart.
+
+For an overview of the Cloud Trace UI, see: [View traces overview].
+
+## OpenTelemetry Dependency
+
+The Google Cloud C++ client libraries use [OpenTelemetry] to provide
+observability into their operation at runtime. To enable tracing,
+`google-cloud-cpp` takes a direct dependency on
+[`opentelemetry-cpp`][opentelemetry-cpp].
+
+OpenTelemetry provides full [install instructions][opentelemetry-cpp-install].
+We will highlight the important points.
+
+### Compatibility with Abseil
+
+OpenTelemetry backports Abseil types, defining them in `namespace absl`. The
+Google Cloud client libraries depends on Abseil, which also defines these types
+in the same namespace. In order to avoid ambiguous symbols, we must set certain
+flags when compiling `opentelemetry-cpp`.
+
+### Testing
+
+The client library is only tested against the latest version of
+`opentelemetry-cpp`. That is v1.9.0, circa 2023-05.
+
+## Using with Bazel
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+1. Install Bazel using [the instructions][bazel-install] from the `bazel.build`
+   website.
+
+1. Compile this example using Bazel:
+
+   ```bash
+   cd $HOME/google-cloud-cpp/google/cloud/opentelemetry/quickstart
+   bazel build ...
+   ```
+
+   Note that Bazel automatically downloads and compiles all dependencies of the
+   project. As it is often the case with C++ libraries, compiling these
+   dependencies may take several minutes.
+
+1. Run the example, changing the placeholders to appropriate values:
+
+   ```bash
+   bazel run :quickstart -- [BUCKET_NAME] [PROJECT_ID]
+   ```
+
+### Details
+
+Note the following two feature flags explicitly set in the `.bazelrc`. Together,
+these flags enable OpenTelemetry tracing instrumentation in `google-cloud-cpp`.
+
+```py
+# Required for OpenTelemetry + Abseil compatibility
+build --@io_opentelemetry_cpp//api:with_abseil
+
+# Enables tracing instrumentation in google-cloud-cpp
+build --@google_cloud_cpp//:enable-experimental-opentelemetry
+```
+
+Also note that we explicitly load OpenTelemetry's dependencies in the
+`WORKSPACE.bazel`.
+
+## Using with CMake
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+### Package Managers
```

Comment:
I would remove this section for now.  Or maybe say "we are working on this".

Reply:
Removed.